Controller/NotifyHandler.php

```diff
@@ -195,16 +196,39 @@ private function _onNotificationSuccess($oResponse, $oBackendService, $sPaymentI
         }
 
         $oOrder = oxNew(Order::class);
+        $bSavedTransaction = self::_saveIfUnmatchedPoiPiaTransaction($sTransactionId, $oResponse, $oBackendService, $oOrder);
+
+        if (!$bSavedTransaction) {
+            ResponseHandler::onSuccessResponse($oResponse, $oBackendService, $oOrder);
+        }
+        return;
+    }
+
+    /**
+     * Saves transaction if it is unmatched and returns true. Otherwise returns false.
+     *
+     * @param string $sTransactionId
+     * @param SuccessResponse $oResponse
+     * @param BackendService  $oBackendService
+     * @param Order $oOrder
+     *
+     * @return boolean
+     *
+     * @since 1.3.0
+     */
+    private function _saveIfUnmatchedPoiPiaTransaction($sTransactionId, $oResponse, $oBackendService, &$oOrder)
+    {
         if ($this->_loadOrder($oOrder, $sTransactionId) >= self::MAX_TIMEOUT_SECONDS) {
```

Comment:
Let's please use an **early exit** instead avoiding nested if-statements:
```
if (($this->_loadOrder($oOrder, $sTransactionId) <= self::MAX_TIMEOUT_SECONDS) || 
($oResponse->getPaymentMethod() !== BasePoiPiaPaymentMethod::PAYMENT_METHOD_WIRETRANSFER)) {
  return false;
}
$this->_oLogger->info('Save unmatched POI/PIA transaction (no related order): ' . $sTransactionId);

ResponseHandler::saveTransaction($oResponse, $oOrder, $oBackendService);
return true;
```

also changed:
1. Timeout check (max timeout still valid)
2. Log as info (instead of error)
3. Remove redundant comment (method name clear enough)